### PR TITLE
Expand WordInfo popover

### PR DESCRIPTION
**Pull request by fsimonjetz**
Add aligned tokens to word popovers (Trello: [Show info for manuscript](https://trello.com/c/70pZysFU))

### CODE REVIEW

File: src/corpus/ui/ChapterViewLine.tsx [hunk 1]
```diff
@@ -286,7 +287,14 @@ export function ChapterViewLineVariant({
   )
 
   return (
-    <>
+    <LineInfoContext.Provider
```
Comment (fsimonjetz) [inline src/corpus/ui/ChapterViewLine.tsx:290]: The popover needs `chapter.id`, `lineNumber`, `variantNumber` and `textService` to fetch corresponding manuscript lines. As the popover is deeply nested and this information should also be shared by `Score` (which is located elsewhere in the tree), it seems reasonable to create it as context.

File: src/corpus/ui/PopoverAlignment.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  | {
+      chapterId: ChapterId
+      lineNumber: number
+      variantNumber: number
+      textService: TextService
+    }
+  | Record<string, never>
+>({})
+
+const AlignedManuscriptTokens = withData<
```
Comment (fsimonjetz) [inline src/corpus/ui/PopoverAlignment.tsx:17]: This is almost identical to `Score`.

File: src/transliteration/ui/WordInfo.tsx [hunk 3]
```diff
@@ -59,7 +62,21 @@ export default function WordInfo({
   tokenClasses: readonly string[]
 }>): JSX.Element {
   const dictionary = useDictionary()
-
+  const { chapterId, lineNumber, variantNumber, textService } = useContext(
+    LineInfoContext
+  )
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const AlignedManuscriptLines = useMemo(
```
Comment (fsimonjetz) [inline src/transliteration/ui/WordInfo.tsx:69]: Creates the list of manuscripts (for now, only shows the `sigla`). Unlike the `Score` component, which is within a `CollapsingRow` that gets created once and is mounted to the DOM, this part is re-rendered every time the popover is opened, incl. fetching the data.

File: src/corpus/ui/PopoverAlignment.tsx (hunk 2 above)
Comment (yCobanoglu) [reply src/corpus/ui/PopoverAlignment.tsx:17]: This is not the place to [useContext](https://reactjs.org/docs/context.html).

File: src/corpus/ui/PopoverAlignment.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          <li key={index}>{manuscript.siglum}</li>
+        ))}
+      </ul>
+    )
+  },
+  ({ id, lineNumber, variantNumber, textService }) =>
+    textService.findChapterLine(id, lineNumber, variantNumber)
+)
+
+export default AlignedManuscriptTokens
```
Comment (yCobanoglu) [inline src/corpus/ui/PopoverAlignment.tsx:40]: export default directly so you don't have to declare a const.

**Review by khoidt — APPROVED** [review]

File: src/corpus/ui/AlignedManuscriptTokens.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+export function createAlignmentMap(
+  manuscripts: ManuscriptLineDisplay[],
+  tokenIndex: number
+): Map<string, AlignedTokenRow> {
+  const map = new Map<string, AlignedTokenRow>()
+
+  for (const manuscript of manuscripts) {
+    const alignedTokens = manuscript.line.content.filter(
+      (token) => _.isNumber(token.alignment) && token.alignment === tokenIndex
+    )
```
Comment (yCobanoglu) [inline src/corpus/ui/AlignedManuscriptTokens.tsx:39]: `token.alignment?  === tokenIndex`

Comment (yCobanoglu) [reply src/corpus/ui/AlignedManuscriptTokens.tsx:39]: can alignedTokens be many instances ?

File: src/corpus/ui/AlignedManuscriptTokens.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+    for (const token of alignedTokens) {
+      const word = token as LemmatizableToken
+      const key = word.value
+      const siglum = manuscript.siglum
+      const isVariant = !_.isNil(word.variant)
+
```
Comment (yCobanoglu) [inline src/corpus/ui/AlignedManuscriptTokens.tsx:46]: map, keine const direkt übergeben und ternary operator

File: src/transliteration/ui/DisplayToken.tsx [hunk 7]
```diff
@@ -265,10 +274,15 @@ function WordComponent({
...
+  showPopover,
 }: TokenProps): JSX.Element {
   const word = token as Word
   return (
-    <WordInfo word={word} tokenClasses={modifierClasses ?? []}>
+    <WordInfo
+      word={word}
+      tokenClasses={modifierClasses ?? []}
+      showPopover={showPopover}
+    >
```
Comment (yCobanoglu) [inline src/transliteration/ui/DisplayToken.tsx:285]: `isPopoverVisible`

File: src/transliteration/ui/DisplayToken.tsx [hunk 8]
```diff
@@ -29,6 +29,8 @@ interface TokenProps {
   token: Token
   Wrapper: TokenWrapper
   tokenClasses?: readonly string[]
+  tokenIndex?: number
+  showPopover?: boolean
```
Comment (yCobanoglu) [inline src/transliteration/ui/DisplayToken.tsx:33]: use `number | null`

File: src/transliteration/ui/DisplayToken.tsx [hunk 9]
```diff
@@ -236,14 +238,21 @@ function LineBreakComponent({ Wrapper }: TokenProps): JSX.Element {
 
 function AkkadianWordComponent({
   token,
+  tokenIndex,
   Wrapper,
```
Comment (yCobanoglu) [inline src/transliteration/ui/DisplayToken.tsx:242]: `=null` if parameter is optional

File: src/transliteration/ui/WordInfo.test.tsx [hunk 10]
```diff
@@ -39,6 +56,53 @@ const word: Word = {
   type: 'Word',
   hasVariantAlignment: false,
 }
+const LINE_INFO = {
+  chapterId: {
```
Comment (yCobanoglu) [inline src/transliteration/ui/WordInfo.test.tsx:60]: lieber klein

File: src/transliteration/ui/WordInfo.tsx [hunk 11]
```diff
@@ -34,32 +37,66 @@ function WordItem({ word }: { word: Word }): JSX.Element {
...
+
+export function fetchLemma({
+  word,
+  dictionary,
+}: {
+  word: LemmatizableToken
+  dictionary: WordService
+}): Bluebird<DictionaryWord[]> {
+  return Bluebird.all(word.uniqueLemma.map((lemma) => dictionary.find(lemma)))
+}
```
Comment (yCobanoglu) [inline src/transliteration/ui/WordInfo.tsx:62]: better to refactor into `WordService` if used in two places.

File: src/corpus/ui/ChapterViewLine.tsx [hunk 12]
```diff
@@ -286,7 +287,14 @@ export function ChapterViewLineVariant({
...
+      value={{
+        chapterId: chapter.id,
+        lineNumber: lineNumber,
+        variantNumber: variantNumber,
+        textService: textService,
+      }}
```
[the remark this answers is not in the document]
Comment (yCobanoglu) [reply src/corpus/ui/ChapterViewLine.tsx:296]: Ein Context reicht, lambda optional

**Pull request closed without merging**